reactor-pool/src/main/java/reactor/pool/SimpleDequePool.java

```diff
@@ -366,7 +366,7 @@ private void drainLoop() {
 						borrower.fail(new PoolShutdownException());
 						return;
 					}
-					borrower.stopPendingCountdown();
+					borrower.stopPendingCountdown(true);
```

Comment:
This is actually followed by `borrower.deliver` which calls `stopPendingCountdown`. Why not rely on the one in `deliver` then? Is it because of the re-scheduling onto the `acquisitionScheduler`? Perhaps there should be an abstraction for signalling that pending state is over and what the next step is so that it is encapsulated? It feels unwieldy that the logic is spread over two classes and feels error-prone. Shouldn't this actually cancel the timeout timer instead?